repl/src/dotty/tools/repl/UserFacingPrinter.scala

```diff
@@ -21,6 +21,7 @@ import dotc.printing.{ GlobalPrec, DotPrec, Printer, PlainPrinter }
 import dotc.typer.Implicits.SearchResult
 import dotc.typer.ImportInfo
 
+// TODO: Avoid code duplication between userfacing and refined printers
```

Comment:
Most of the duplication is gone in https://github.com/lampepfl/dotty/pull/3032 so you should be able to drop this commit when rebasing.